bbot/core/event/base.py

```diff
@@ -1124,6 +1147,40 @@ class URL_HINT(URL_UNVERIFIED):
     pass
 
 
+class WEB_PARAMETER(DictHostEvent):
+
+    def _data_id(self):
+
+        # dedupe by url:name:param_type
+        url = self.data.get("url", "")
+        name = self.data.get("name", "")
+        param_type = self.data.get("type", "")
+        # REMOVE
+        # this is a hack which needs to be replaced with a real fix in bbot-2.0 branch
+        if self.scan is not None:
+            if self.scan.config.get("url_querystring_remove", True) == False:
+                from urllib.parse import urlparse
+
+                parsed_url = urlparse(url)
+
+                if self.scan.config.get("url_querystring_collapse", True) == True:
+                    # Only consider parameter names in dedup (collapse values)
+                    cleaned_query = "|".join(sorted([p.split("=")[0] for p in parsed_url.query.split("&")]))
```

Comment:
We should avoid manually parsing query strings and use `parse_qs()` instead.

Reply:
https://github.com/blacklanternsecurity/bbot/pull/1528